code/modules/robotics/bot/secbot.dm

```diff
@@ -420,28 +421,36 @@ Report Arrests: <A href='?src=\ref[src];operation=report'>[report_arrests ? "On"
 		return 0
 
 	proc/baton_attack(var/mob/living/carbon/M)
+		sleep(BATON_INITIAL_DELAY)
+		if(!IN_RANGE(src, src.target, 1))
+			return
+
 		src.icon_state = "secbot-c[src.emagged >= 2 ? "-wild" : null]"
 		var/maxstuns = 4
 		var/stuncount = (src.emagged >= 2) ? rand(5,10) : 1
 
 		last_attack = world.time
 
+		// No need for unnecessary hassle, just make it ignore charges entirely for the time being.
+		if (src.our_baton && istype(src.our_baton))
+			if (src.our_baton.uses_electricity == 0)
+				src.our_baton.uses_electricity = 1
+			if (src.our_baton.uses_charges != 0)
+				src.our_baton.uses_charges = 0
+		else
+			src.our_baton = new our_baton_type(src)
```

Comment:
should be safe to keep outside the loop, no need to redo these checks over and over